### PR TITLE
Configure model checkpointing

### PR DESCRIPTION
**Pull request by weiji14**
## What I am changing

- Configuring how model weights are saved to checkpoint (*.ckpt) files during training

## How I did it

- Using Lightning's [ModelCheckpoint](https://lightning.ai/docs/pytorch/2.1.0/api/lightning.pytorch.callbacks.ModelCheckpoint.html) callback with the following parameters:
  - auto_insert_metric_name=False
  - filename="vit_epoch-{epoch:02d}_train_loss-{train/loss:.2f}"
  - monitor="train/loss"
  - mode="min"
  - save_last=True
  - save_top_k=3
  - save_weights_only=True
- Added the [AsyncCheckpointIO](https://lightning.ai/docs/pytorch/2.1.0/api/lightning.pytorch.plugins.io.AsyncCheckpointIO.html) plugin to reduce the gap in time between epochs when a checkpoint is saved

## How you can test it

- Run `python trainer.py fit --trainer.max_epochs=20` locally
- Extra configuration options can be found using `python trainer.py fit --trainer.callbacks.help=ModelCheckpoint`

## Related Issues

References:
- https://github.com/Lightning-AI/lightning/issues/11561

### CODE REVIEW

**Comment by weiji14** [comment]
Gonna merge this in directly since it's a relatively small change, but useful in terms of showing how [callbacks](https://lightning.ai/docs/pytorch/2.1.0/api_references.html#callbacks) and [plugins](https://lightning.ai/docs/pytorch/2.1.0/api_references.html#plugins) can be added to the LightningCLI in trainer.py.
